app.py

```
@@ -0,0 +1,861 @@
+        :param user_comments: Comments entered by the user at export start.
+        :param make_shot_thumb: Should a thumbnail be uploaded to the associated shot as well?
+        """
+        
+        path = os.path.join(info.get("destinationPath"), info.get("resolvedPath"))
+        context = sgtk.context.deserialize(serialized_context)
+        version_number = int(info["versionNumber"])
+        
+        if info.get("assetType") == "video":
+            
+            # first register a publish record in Shotgun for the plates
+            sg_data = self._sg_submit_helper.register_video_publish(context,
+                                                                    path,
+                                                                    user_comments,
+                                                                    version_number,
+                                                                    info["width"],
+                                                                    info["height"],
+                                                                    make_shot_thumb)
+
+            # now create a version record, generate a quicktime and upload it            
+            self._sg_submit_helper.create_version(context,
+                                                  path,
+                                                  user_comments,
+                                                  sg_data,
+                                                  info["width"],
+                                                  info["height"],
+                                                  info["aspectRatio"])
+            
+        elif info.get("assetType") == "batch":
+            
+            # register a publish record in Shotgun for the batch file
+            self._sg_submit_helper.register_batch_publish(context, path, user_comments, version_number)
+                        
+        else:
```

I think in other places we silently return when an unexpected assetType is received, any reason why we are raising an exception here ?